### PR TITLE
Fix yugabyte tablegroups in colocated database

### PR DESCRIPTION
**Pull request by ZhengLin-Li**
fix: https://github.com/sqlancer/sqlancer/issues/789

In this PR, we fix the issue that SQLancer may create tablegroups in colocated database.

### CODE REVIEW

File: src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java
Comment (bajinsheng) [inline src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java:161]: Is this error caused by the same reason? Can we remove this error?

File: src/sqlancer/yugabyte/ysql/gen/YSQLTableGroupGenerator.java
Comment (bajinsheng) [inline src/sqlancer/yugabyte/ysql/gen/YSQLTableGroupGenerator.java:18]: Remove this empty line

File: src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java
Comment (bajinsheng) [inline src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java:162]: Why we need this try catch?

File: src/sqlancer/yugabyte/ysql/YSQLSchema.java
Comment (mrigger) [inline src/sqlancer/yugabyte/ysql/YSQLSchema.java:167]: Can it actually be that the database system returns an empty result, or would it be a bug in the database system? If it would be a bug, we should not ignore it.

File: src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java
Comment (mrigger) [reply src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java:162]: I think a cleaner way would be to let `getDatabaseIsColocated` return `Optional` (in case an empty result set might be expected for the query that `getDatabaseIsColocated` executes).

File: src/sqlancer/yugabyte/ysql/YSQLSchema.java
Comment (ZhengLin-Li) [reply src/sqlancer/yugabyte/ysql/YSQLSchema.java:167]: I think actually it won't returns an empty result. See: https://www.yugabyte.com/blog/differences-primary-key-usage-colocated-noncolocated-tables/#:~:text=a%20colocated%20database%3A-,SELECT%20yb_is_database_colocated()%3B,-Primary%20key%20syntaxes

Also I test it with and without selecting a databse, and it turned out to return a result.

File: src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java
Comment (ZhengLin-Li) [reply src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java:161]: I think it is not the same error. 

What we are tring to fix is: `ERROR: cannot use tablegroups in a colocated database`. see: https://github.com/sqlancer/sqlancer/issues/789

But here it is `Cannot use TABLEGROUP with TEMP table`

File: src/sqlancer/yugabyte/ysql/YSQLSchema.java
Comment (mrigger) [reply src/sqlancer/yugabyte/ysql/YSQLSchema.java:167]: In this case, I think you can remove the check, and thus also remove the exception-handling code in the caller.

File: src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java
Comment (ZhengLin-Li) [reply src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java:162]: What does it mean by saying return `optional`. Could give a example?

File: src/sqlancer/yugabyte/ysql/YSQLSchema.java
Comment (ZhengLin-Li) [reply src/sqlancer/yugabyte/ysql/YSQLSchema.java:167]: Ok, make sense.

File: src/sqlancer/yugabyte/ysql/YSQLSchema.java
Comment (mrigger) [inline src/sqlancer/yugabyte/ysql/YSQLSchema.java:166]: I guess we can remove the check here, right?

File: src/sqlancer/yugabyte/ysql/YSQLSchema.java
Comment (mrigger) [inline src/sqlancer/yugabyte/ysql/YSQLSchema.java:171]: Typically, we throw an `AssertionError`, but I think this should be fine as well

**Comment by mrigger** [comment]
LGTM, after fixing the PMD warning we can merge the PR.

**Review by bajinsheng — APPROVED** [review]
LGTM! But it seems this branch is out of date?

**Comment by mrigger** [comment]
@bajinsheng and I briefly chatted about this - since the change is unlikely to break anything despite being outdated with respect to the main branch, we decided to merge the PR (currently, it seems only I have the necessary permissions).

**Pull request merged**
